src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1570,6 +1585,8 @@ impl Disambiguator {
     fn ns(self) -> Namespace {
         match self {
             Self::Namespace(n) => n,
+            // for purposes of link resolution, fields are in the value namespace.
+            Self::Kind(DefKind::Field) => ValueNS,
```

Comment:
i'm confused, i thought you were going to remove this? is the problem that `DefKind::ns` is defined in rustc so you can't modify it?

Reply:
I was, but it turns out it didn't make sense to move the `disambiguator == Field` so far up and out - that would mean putting it in `resolve_disambiguator()`, while it makes much more sense (and is easier to keep existing field-last resolution behavior) if it just gets considered to be in the value ns and fails to resolve to anything before getting picked up in the special-case in `resolve_associated_item()`.